res/controllers/Roland_DJ-505-scripts.js

```diff
@@ -1094,59 +1094,63 @@ DJ505.PadSection.prototype.padModeButtonPressed = function (channel, control, va
 };
 
 DJ505.PadSection.prototype.paramButtonPressed = function (channel, control, value, status, group) {
-    var mode = this.controlToPadMode(this.padMode);
-    if (mode && mode.paramPlusButton && mode.paramMinusButton) {
+    if (this.currentMode && this.currentMode.paramPlusButton && this.currentMode.paramMinusButton) {
         if (control & 1) {
-            mode.paramPlusButton.input(channel, control, value, status, group);
+            this.currentMode.paramPlusButton.input(channel, control, value, status, group);
         } else {
-            mode.paramMinusButton.input(channel, control, value, status, group);
+            this.currentMode.paramMinusButton.input(channel, control, value, status, group);
         }
     }
 };
 
 DJ505.PadSection.prototype.setPadMode = function (control) {
-    if (this.padMode === control) {
-        return;
-    }
-
     var mode = this.controlToPadMode(control);
-    if (mode === undefined) {
+    if (this.currentMode === mode || mode === undefined) {
```

Comment:
Should this be
```suggestion
    if (this.currentMode === mode || this.currentMode === undefined) {
```
?

Reply:
No. `mode` is the *new* mode. If it's `undefined`, then the user pressed a mode button that is currently not mapped (e.g. `FLIP` mode, `SLICER` mode, etc.). In that case, nothing happens and the current mode stays active. I'll add a commit that clears this up by using better variable names.